game-03/components/player/player.tscn

```diff
@@ -15,116 +600,504 @@ height = 92.0
+script = ExtResource("4_mcw5b")
+name = &"else"
+children = Array[ExtResource("4_mcw5b")]([SubResource("Resource_vg0qd"), SubResource("Resource_invbt")])
+arguments = {}
+
+[sub_resource type="Resource" id="Resource_wshyy"]
+script = ExtResource("4_mcw5b")
+name = &"process"
+children = Array[ExtResource("4_mcw5b")]([SubResource("Resource_1au78"), SubResource("Resource_boej2"), SubResource("Resource_4vbry"), SubResource("Resource_nfx6g"), SubResource("Resource_ql1hb")])
+arguments = {}
+
+[sub_resource type="Resource" id="Resource_widc7"]
+script = ExtResource("4_ckbst")
+root = SubResource("Resource_wshyy")
+canvas_position = Vector2(25, 100)
+
+[sub_resource type="Resource" id="Resource_43f8x"]
+script = ExtResource("6_c3puh")
+name = &"get_node"
+arguments = {
+"path": NodePath("AnimationPlayer")
+}
+
+[sub_resource type="Resource" id="Resource_cskn5"]
+script = ExtResource("4_mcw5b")
+name = &"call_method_node"
+children = Array[ExtResource("4_mcw5b")]([])
+arguments = {
+"method_name": "be_hit",
+"node": SubResource("Resource_43f8x")
+}
+
+[sub_resource type="Resource" id="Resource_apvp5"]
+script = ExtResource("6_c3puh")
+name = &"vector2"
+arguments = {
+"x": 512.0,
+"y": 448.0
+}
+
+[sub_resource type="Resource" id="Resource_32s8y"]
+script = ExtResource("4_mcw5b")
+name = &"Node2D_set_position"
+children = Array[ExtResource("4_mcw5b")]([])
+arguments = {
+"value": SubResource("Resource_apvp5")
+}
+
+[sub_resource type="Resource" id="Resource_hv1r0"]
+script = ExtResource("4_mcw5b")
+name = &"define_method"
+children = Array[ExtResource("4_mcw5b")]([SubResource("Resource_cskn5"), SubResource("Resource_32s8y")])
+arguments = {
+"method_name": &"reset_player"
+}
+
+[sub_resource type="Resource" id="Resource_gsl8k"]
+script = ExtResource("4_ckbst")
+root = SubResource("Resource_hv1r0")
+canvas_position = Vector2(375, 0)
+
+[sub_resource type="Resource" id="Resource_hsh3i"]
+script = ExtResource("7_ubegq")
+var_name = "is_playing"
+var_type = 1
+
+[sub_resource type="Resource" id="Resource_ahm1y"]
 script = ExtResource("6_eqqyd")
-script_inherits = "Area2D"
-block_trees = Array[ExtResource("4_mcw5b")]([SubResource("Resource_sxoca")])
-variables = Array[ExtResource("7_uyqn1")]([])
-generated_script = "extends Area2D
+script_inherits = "SimpleCharacter"
+block_serialization_trees = Array[ExtResource("4_ckbst")]([SubResource("Resource_xjswj"), SubResource("Resource_widc7"), SubResource("Resource_gsl8k")])
+variables = Array[ExtResource("7_ubegq")]([SubResource("Resource_hsh3i")])
+generated_script = "extends SimpleCharacter
 
+var is_playing: bool
 
-func _on_body_entered(body: Node2D):
 
-	if body.is_in_group('enemy'):
-		body.call('vanish_enemy')
+func _ready():
+	is_playing = false
+
+func _process(delta):
+	if (Input.is_action_pressed('ui_accept')):
+		is_playing = true
+	if (is_playing):
+		move_with_player_buttons('player_1', 'platformer', delta)
+	if ((((velocity).x) == 0) and (((velocity).y) == 0)):
```

Comment:
Instead of writing this:

![if x of velocity == 0 and y of velocity == 0](https://github.com/user-attachments/assets/e2710cb8-5f6a-440c-a2a8-d6378c7b549a)

You could also write:

![if not velocity](https://github.com/user-attachments/assets/2769c28f-85b5-4652-adb9-38610e95ab03)

since this change:

https://github.com/endlessm/godot-block-coding/commit/80a8fc29c9b7fcdc377449b9653574e6a09574b9

However I think what you have here is clearer.

Reply:
Good point!  I even thought getting the length of the vector.  But, ya, get the logic simpler and clearer.